lib/lev/status.rb

```diff
@@ -2,93 +2,142 @@
 
 module Lev
   class Status
+    attr_reader :id, :status, :progress, :errors
+
     STATE_QUEUED = 'queued'
     STATE_WORKING = 'working'
     STATE_COMPLETED = 'completed'
     STATE_FAILED = 'failed'
     STATE_KILLED = 'killed'
+    STATE_UNKNOWN = 'unknown'
 
     STATES = [
       STATE_QUEUED,
       STATE_WORKING,
       STATE_COMPLETED,
       STATE_FAILED,
-      STATE_KILLED
+      STATE_KILLED,
+      STATE_UNKNOWN
     ].freeze
 
-    attr_reader :uuid
+    def initialize(attrs = {})
+      @id = attrs[:id] || attrs['id'] || SecureRandom.uuid
+      @status = attrs[:status] || attrs['status'] || STATE_QUEUED
```

Comment:
initial state should be `STATE_UNKNOWN`.  Only when the status is `queued!` should its state change (which admittedly happens in routine.rb immediately after initialization)